File: app/src/main/java/com/example/myapplication/models/mangamodels/DetailMangaModel.java
package com.example.myapplication.models.mangamodels;

import java.util.List;

import lombok.Data;

@Data
public class DetailMangaModel {
    private String mangaTitle;
    private String mangaThumb;
    private String otherNames;
    private String mangaAuthor;
    private String mangaSynopsis;
    private String mangaStatus;
    private String mangaType;
    private String mangaRating;
    private String firstUpdateYear;
    private String totalMangaChapter;
    private String lastMangaUpdateDate;
    private List<DetailMangaGenres> detailGenres;
    private List<DetailAllChapterDatas> allChapterDatas;

    @Data
    public static class DetailAllChapterDatas {
        private String chapterTitle;
        private String chapterURL;
        private String chapterReleaseTime;
    }

    @Data
    public static class DetailMangaGenres {
        private String genreTitle;
        private String genreURL;
    }
}

File: app/src/main/java/com/example/myapplication/models/mangamodels/DetailMangaModel.kt
package com.example.myapplication.models.mangamodels

data class DetailMangaModel(
        var mangaTitle: String? = null,
        var mangaThumb: String? = null,
        var otherNames: String? = null,
        var mangaAuthor: String? = null,
        var mangaSynopsis: String? = null,
        var mangaStatus: String? = null,
        var mangaType: String? = null,
        var mangaRating: String? = null,
        var firstUpdateYear: String? = null,
        var totalMangaChapter: String? = null,
        var lastMangaUpdateDate: String? = null,
        var detailGenres: List<DetailMangaGenres>? = null,
        var allChapterDatas: List<DetailAllChapterDatas>? = null
) {
    data class DetailAllChapterDatas(
            var chapterTitle: String? = null,
            var chapterURL: String? = null,
            var chapterReleaseTime: String? = null
    )

    data class DetailMangaGenres(
            var genreTitle: String? = null,
            var genreURL: String? = null
    )
}
